spec/support/parser_examples.rb

```diff
@@ -0,0 +1,39 @@
+shared_examples_for 'a traffic source campaign parser' do
+
+  it 'should have campaign correctly set' do
+    expect(subject.campaign).to eq campaign
+  end
+
+  it 'should have medium correctly set' do
+    expect(subject.medium).to eq medium
+  end
+
+  it 'should have source correctly set' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.